spec/controllers/media_objects_controller_spec.rb

```diff
@@ -279,6 +279,38 @@
           expect(new_media_object.other_identifier.find {|id_pair| id_pair[:source] == 'other'}).not_to be nil
           expect(new_media_object.other_identifier.find {|id_pair| id_pair[:source] == 'other'}[:id]).to eq('12345')
         end
+        it "should create a new media_object using ingest_api_hash of existing media_object" do
+          # master_file_obj = FactoryGirl.create(:master_file, master_file.slice(:files))
+          media_object = FactoryGirl.create(:fully_searchable_media_object)
+          master_file = FactoryGirl.create(:master_file, :with_derivative, :with_thumbnail, :with_poster, :with_structure, :with_captions)
+          media_object.ordered_master_files += [master_file]
+          media_object.workflow.last_completed_step = 'access-control'
```

Comment:
Should this get included in the media object factory?

Reply:
I don't think so. This is specifically for the API to have a fully-ingested object. I don't know, but I think our ingest workflow tests that require partially ingested objects.

Reply:
Looks like we have a trait which captures this: `:with_completed_workflow`.